controllers/che/checluster_controller.go

```diff
@@ -253,10 +248,30 @@ func (r *CheClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	deployContext := &deploy.DeployContext{
 		ClusterAPI: clusterAPI,
-		CheCluster: instance,
+		CheCluster: checluster,
+	}
+
+	result, err := r.doReconcile(deployContext)
+	if err != nil {
+		logrus.Errorf("Reconciliation failed, cause: %v", err)
+		if err := deploy.SetStatusDetails(deployContext, failedValidationReason, err.Error(), ""); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
+		logrus.Info("Reconciliation completed.")
+		if err := deploy.SetStatusDetails(deployContext, "", "", ""); err != nil {
```

Comment:
Could you check, if we need to update status before submitting an update?
Because in most cases it will update empty to empty.